jupyterhubutils/spawner/lsstspawner.py

```diff
@@ -0,0 +1,397 @@
+                    if tag == "__custom":
+                        cit = self.user_options.get('image_tag')
+                        if cit:
+                            image = imgname + ":" + cit
+                self.log.info("Replacing image from options form: %s" % image)
+                size = self.user_options.get('size')
+                if size:
+                    image_size = om._sizemap[size]
+                clear_dotlocal = self.user_options.get('clear_dotlocal')
+        else:
+            self.log.warning("No user options found.  That seems wrong.")
+        if clear_dotlocal:
+            pod_env['CLEAR_DOTLOCAL'] = "TRUE"
+            em.set_env("CLEAR_DOTLOCAL", "TRUE")
+        # Set up Lab pod resource constraints (not namespace quotas)
+        mem_limit = em.get_env_key('LAB_MEM_LIMIT')
+        cpu_limit = em.get_env_key('LAB_CPU_LIMIT')
+        if image_size:
+            mem_limit = str(int(image_size["mem"])) + "M"
+            cpu_limit = image_size["cpu"]
+        cpu_limit = float(cpu_limit)
+        self.mem_limit = mem_limit
+        self.cpu_limit = cpu_limit
+        mem_guar = em.get_env_key('LAB_MEM_GUARANTEE')
+        if mem_guar is None:
+            mem_guar = '1M'
+        cpu_guar = em.get_env_key('LAB_CPU_GUARANTEE')
+        if cpu_guar is None:
+            cpu_guar = 0.02
+        mem_guar = str(mem_guar)
+        cpu_guar = float(cpu_guar)
+        # Tiny gets the "basically nothing" above (or the explicit
+        #  guarantee).  All others get 1/LAB_SIZE_RANGE times their
+        #  maximum, with a default of 1/4.
+        size_range = em.get_env_key('LAB_SIZE_RANGE')
+        if size_range is None:
+            size_range = 4.0
+        size_range = float(size_range)
+        if image_size and size != 'tiny':
+            mem_guar = int(image_size["mem"] / size_range)
+            cpu_guar = float(image_size["cpu"] / size_range)
+        self.mem_guarantee = mem_guar
+        self.cpu_guarantee = cpu_guar
+        # Figure out the image and set the pod name from it.
+        self.log.debug("Image: {}".format(image))
+        pod_env['JUPYTER_IMAGE_SPEC'] = image
+        em.set_env('JUPYTER_IMAGE_SPEC', image)
+        self.image = image
+        # Parse the image name + tag
+        i_l = image.split("/")
+        if len(i_l) == 1:
+            repo_tag = i_l[0]
+        else:
+            repo_tag = i_l[1]
+        repo = repo_tag.split(":")[0]
+        rt_tag = tag.replace('_', '-')
+        abbr_pn = repo
+        if repo == 'sciplat-lab':
+            # Saving characters because tags can be long
+            abbr_pn = "nb"
+        pn_template = abbr_pn + "-{username}-" + rt_tag
+        pod_name = self._expand_user_properties(pn_template)
+        self.pod_name = pod_name
+        self.log.info("Replacing pod name from options form: %s" %
+                      pod_name)
+        # Get volume definitions from volume manager.
```

Comment:
So we are in the SPAWNING phase here, so run ensure_namespaces and all the stuff that makes stuff here.

Or even better, make a function called make_stuff()

Reply:
LSSTMiddleManager.make_stuff()

Reply:
ensure_resources() and it's called in spawner.start()